daml-assistant/src/DAML/Assistant/Install.hs

```diff
@@ -173,10 +184,44 @@ activateDaml env@InstallEnv{..} targetPath = do
                      ]
             else createSymbolicLink damlBinarySourcePath damlBinaryTargetPath
 
-    unlessQuiet env $ do -- Ask user to add .daml/bin to PATH if it is absent.
-        searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
-        when (damlBinaryTargetDir `notElem` searchPaths) $ do
-            output ("Please add " <> damlBinaryTargetDir <> " to your PATH.")
+    updatePath env damlBinaryTargetDir
+
+updatePath :: InstallEnv -> FilePath -> IO ()
+#ifdef mingw32_HOST_OS
+updatePath env@InstallEnv{..} targetPath = do
+    -- Updating PATH on Windows is annoying so we do it automatically.
+    bracket (regOpenKeyEx hKEY_CURRENT_USER "Environment" kEY_ALL_ACCESS) regCloseKey $ \envKey -> do
+        path <- regQueryValueEx envKey "Path"
+        let newPath = targetPath <> ";" <> remove targetPath path
```

Comment:
In the (very) corner case of having no Path you end up with a redundant trailing `;`. Could we append to the list then intercalate at the end?

Reply:
Good catch, fixed